src/crimson/mon/MonClient.cc

```diff
@@ -816,7 +826,11 @@ seastar::future<> Client::handle_auth_reply(crimson::net::ConnectionRef conn,
   if (found != pending_conns.end()) {
     return (*found)->handle_auth_reply(m);
   } else if (active_con) {
-    return active_con->handle_auth_reply(m);
+    return active_con->handle_auth_reply(m).then([this] {
+      return seastar::when_all_succeed(
+        active_con->renew_rotating_keyring(),
+        active_con->renew_tickets()).then_unpack([] {});
```

Comment:
```suggestion
        active_con->renew_tickets())..discard_result();
```

Reply:
I guess this can be applied in other places too. If so, will address as a follow up / separate commit.

Reply:
yup. we can have a follow-up PR to implement this cleanup.

Reply:
Sent as https://github.com/ceph/ceph/pull/47285.